### DISCUSSION TITLE
ATTiny 1616, MTC 2.6.5, Optimization problem?

### DISCUSSION THREAD

**Discussion by unhinged-engineer** [SpenceKonde/megaTinyCore#915] [Q&A]
Please be kind, this is my first ever post on github, so I really have no idea what I'm doing here.

I'm trying to write code for a Tiny 1616, and just getting started. The project is going to be processing speed critical, so I'm trying to use -O3, instead of -OS for the optimization.

When I use -OS, it compiles fine, but when I use -O3, I get the following error:

C:\Users\jrector\AppData\Local\Temp\ccIg8Jos.s: Assembler messages:
C:\Users\jrector\AppData\Local\Temp\ccIg8Jos.s:454: Error: symbol `_poll_dre_done' is already defined
C:\Users\jrector\AppData\Local\Temp\ccIg8Jos.s:728: Error: symbol `_poll_dre_done' is already defined
lto-wrapper.exe: fatal error: C:\Users\jrector\AppData\Local\Arduino15\packages\DxCore\tools\avr-gcc\7.3.0-atmel3.6.1-azduino6/bin/avr-gcc returned 1 exit status
compilation terminated.
c:/users/jrector/appdata/local/arduino15/packages/dxcore/tools/avr-gcc/7.3.0-atmel3.6.1-azduino6/bin/../lib/gcc/avr/7.3.0/../../../../avr/bin/ld.exe: error: lto-wrapper failed
collect2.exe: error: ld returned 1 exit status

exit status 1

Compilation error: exit status 1

The code that I'm trying to compile is VERY simple, for now, because I'm just getting started.

`void setup() {
  //SPI.swap(1);
  Serial.swap(1);
  Serial.begin(512000, SERIAL_8N1 | SERIAL_HALF_DUPLEX | SERIAL_TX_ONLY);
}

void loop() {
  Serial.print("hello world");
  delay(1000);
}`

for some reason my code formatting isn't working correctly, because of the error? I think you can see what I'm doing, regardless.

Upshot: Is there some setting or command that I'm doing wrong, to make the error happen, but only when -O3 optimized?

**Comment by SpenceKonde** [comment]
Moving to issues as this is a bug and locking this topic to keep all discussion in the same place. This issue will also impact DxCore as well. Please direct further comments to that issue.
